src/main/java/org/jboss/staxmapper/XMLElementReaderProvider.java

```diff
@@ -0,0 +1,18 @@
+package org.jboss.staxmapper;
+
+/**
+ * Provider for {@link XMLElementReader} that serves as layer to allow lazy parsers creation only when they are needed.
+ * @author Tomaz Cerar (c) 2016 Red Hat Inc.
+ */
+@FunctionalInterface
+public interface XMLElementReaderProvider<T> {
```

Comment:
Is there a good reason to have a separate type for this instead of just using a ``Supplier<XMLElementReader<T>>``?